source/distributed/shared_tria.cc

```diff
@@ -134,6 +135,20 @@ namespace parallel
       this->update_number_cache ();
     }
 
+    template <int dim, int spacedim>
+    void
+    Triangulation<dim, spacedim>::
+    copy_triangulation (const dealii::Triangulation<dim, spacedim> &old_tria)
+    {
+      Assert ((dynamic_cast<const dealii::parallel::distributed::Triangulation<dim,spacedim> *>(&old_tria) == NULL),
+              ExcMessage("Cannot use this function on parallel::distributed::Triangulation."));
+
+      dealii::parallel::Triangulation<dim,spacedim>::copy_triangulation (old_tria);
+      dealii::Triangulation<dim,spacedim>::copy_triangulation (old_tria);
```

Comment:
it currently does not because the logic of p::d::Tria::copy_triangulation(), which is roughly 
`::Triangulation<dim,spacedim>::copy_triangulation (old_tria);` then do something, then if we copying from `p::d::Tria`, call `::parallel::Triangulation<dim,spacedim>::copy_triangulation (old_tria);`.